src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+    private final int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int threshold = 12;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int hash = hash(key);
+        Node<K, V> newEntry = new Node<>(hash, key, value, null);
+        if (table[hash] == null) {
+            table[hash] = newEntry;
+        } else {
+            Node<K, V> previousNode = null;
+            Node<K, V> currentNode = table[hash];
+            while (currentNode != null) {
+                if (currentNode.key == null && key != null) {
+                    previousNode = currentNode;
+                    currentNode = currentNode.next;
+                    continue;
+                }
+                if (key == null && currentNode.key == null) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.key.equals(key)) {
+                    newEntry.next = currentNode.next;
+                    if (previousNode == null) {
+                        table[hash] = newEntry;
+                    } else {
+                        previousNode.next = newEntry;
+                    }
+                    return;
+                }
+                previousNode = currentNode;
+                currentNode = currentNode.next;
+            }
+            previousNode.next = newEntry;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
```

Comment:
It's difficult to read and understand this method. Please try to simplify it. Try not to use nested "if"